astroid/interpreter/objectmodel.py

```diff
@@ -118,6 +118,19 @@ def lookup(self, name):
             return getattr(self, IMPL_PREFIX + name)
         raise AttributeInferenceError(target=self._instance, attribute=name)
 
+    @property
+    def attr___new__(self):
+        """Calling cls.__new__(cls) on an object returns an instance of that object.
+
+        Instance is either an instance or a class definition of the instance to be
+        created.
+        """
```

Comment:
I'm wondering if this obviates most of `UnboundMethod.infer_call_result`? Or no?

Reply:
```python
import astroid

mod = astroid.parse(
    """
class A:
    ...

A().__new__(A)
A.__new__(A)
    """
)
list(mod.body[-1].value.infer())
list(mod.body[-2].value.infer())
```

With:
```diff
diff --git a/astroid/bases.py b/astroid/bases.py
index e41964341..dd1e9d2c1 100644
--- a/astroid/bases.py
+++ b/astroid/bases.py
@@ -401,6 +401,7 @@ class UnboundMethod(Proxy):
             qname = self._proxied.parent.frame(future=True).qname()
             # Avoid checking builtins.type: _infer_type_new_call() does more validation
             if qname.startswith("builtins.") and qname != "builtins.type":
+                print("CALLED")
                 return self._infer_builtin_new(caller, context)
         return self._proxied.infer_call_result(caller, context)
 
diff --git a/astroid/interpreter/objectmodel.py b/astroid/interpreter/objectmodel.py
index d120895aa..65be429ce 100644
--- a/astroid/interpreter/objectmodel.py
+++ b/astroid/interpreter/objectmodel.py
@@ -127,6 +127,7 @@ class ObjectModel:
         """
         # TODO: Use isinstance instead of try ... except after _instance has typing
         try:
+            print("CALLED")
             return self._instance._proxied.instantiate_class()
         except AttributeError:
             return self._instance.instantiate_class()
```

Shows that this is used for bound methods. Both are being called in other instances I think.